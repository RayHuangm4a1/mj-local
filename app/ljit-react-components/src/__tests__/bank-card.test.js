import React from 'react';
import { shallow, mount, } from 'enzyme';
import BankCard from '../components/bank-card';

describe('BankCard', () => {
	it('should handle default props', () => {
		const {
			className,
			isUnbindButtonVisible,
			onClickUnbind,
		} = BankCard.defaultProps;

		expect(className).toEqual('');
		expect(isUnbindButtonVisible).toBe(false);
		expect(onClickUnbind).toBeDefined();
		expect(onClickUnbind).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			className: 'mock-class',
			dataSource: {
				bankName: 'mock-bankName',
				payer: 'mock-payer',
				number: 'mock-number',
				activatedAt: '2019-08-13T10:38:43',
				withdrawableAt: '2019-08-13T16:38:43',
			},
			onClickUnbind: () => {},
			isUnbindButtonVisible: true,
		};
		const wrapper = shallow(<BankCard {...props} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-bank-card', () => {
		const wrapper = shallow(
			<BankCard
				dataSource={{
					bankName: 'mock-bankName',
					payer: 'mock-payer',
					number: 'mock-number',
					activatedAt: '2019-08-13T10:38:43',
					withdrawableAt: '2019-08-13T16:38:43',
				}}
			/>
		);

		expect(wrapper.hasClass('ljit-bank-card')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(
			<BankCard
				className={className}
				dataSource={{
					bankName: 'mock-bankName',
					payer: 'mock-payer',
					number: 'mock-number',
					activatedAt: '2019-08-13T10:38:43',
					withdrawableAt: '2019-08-13T16:38:43',
				}}
			/>
		);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	describe('when pass isUnbindButtonVisible', () => {
		it('should has class "ljit-bank-card__bottom class"', () => {
			const wrapper = mount(
				<BankCard
					dataSource={{
						bankName: 'mock-bankName',
						payer: 'mock-payer',
						number: 'mock-number',
						activatedAt: '2019-08-13T10:38:43',
						withdrawableAt: '2019-08-13T16:38:43',
					}}
					isUnbindButtonVisible={true}
				/>
			);

			expect(wrapper.find('.ljit-bank-card__bottom')).toHaveLength(1);
		});
		it('should handle onClickUnbind event', () => {
			const onClickUnbind = jest.fn();
			const dataSource = {
				bankName: 'mock-bankName',
				payer: 'mock-payer',
				number: 'mock-number',
				activatedAt: '2019-08-13T10:38:43',
				withdrawableAt: '2019-08-13T16:38:43',
			};
			const wrapper = mount(
				<BankCard
					onClickUnbind={onClickUnbind}
					dataSource={dataSource}
					isUnbindButtonVisible={true}
				/>
			);

			wrapper.find('button').simulate('click');
			expect(onClickUnbind).toHaveBeenCalledTimes(1);
			expect(onClickUnbind).toHaveBeenCalledWith(dataSource);
		});
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-class',
			dataSource: {
				bankName: 'mock-bankName',
				payer: 'mock-payer',
				number: 'mock-number',
				activatedAt: '2019-08-13T10:38:43',
				withdrawableAt: '2019-08-13T16:38:43',
			},
			isUnbindButtonVisible: true,
			onClickUnbind: () => {},
		};
		const wrapper = mount(<BankCard {...props}/>);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().dataSource).toEqual(props.dataSource);
		expect(wrapper.props().isUnbindButtonVisible).toEqual(props.isUnbindButtonVisible);
		expect(wrapper.props().onClickUnbind).toEqual(props.onClickUnbind);
	});
});
