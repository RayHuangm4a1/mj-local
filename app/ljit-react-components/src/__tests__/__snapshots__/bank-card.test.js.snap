// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankCard should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BankCard
    className="mock-class"
    dataSource={
      Object {
        "activatedAt": "2019-08-13T10:38:43",
        "bankName": "mock-bankName",
        "number": "mock-number",
        "payer": "mock-payer",
        "withdrawableAt": "2019-08-13T16:38:43",
      }
    }
    isUnbindButtonVisible={true}
    onClickUnbind={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bottom": <div
        className="ljit-bank-card__bottom"
      >
        <Button
          color=""
          disabled={false}
          isFullWidth={true}
          onClick={[Function]}
          outline="solid"
        >
          解除绑定
        </Button>
      </div>,
      "className": "ljit-bank-card mock-class",
      "isShowingCloseButton": false,
      "left": <div
        className="ljit-bank-card__left"
      >
        <div
          className="ljit-bank-card__title"
        >
          mock-bankName
        </div>
        <LabelText
          fontSize="small"
          isFixedWidth={true}
          isSpaceBetween={false}
          label="持卡人："
          labelColType="medium"
          labelColor="grey"
          text="mock-payer"
          textColor="black"
        />
        <LabelText
          fontSize="small"
          isFixedWidth={true}
          isSpaceBetween={false}
          label="尾号："
          labelColType="medium"
          labelColor="grey"
          text="mock-number"
          textColor="black"
        />
      </div>,
      "onClose": [Function],
      "right": <div
        className="ljit-bank-card__right"
      >
        <LabelText
          fontSize="small"
          isFixedWidth={true}
          isSpaceBetween={false}
          label="绑定时间"
          labelColType="medium"
          labelColor="grey"
          text="2019/08/13 10:38:43"
          textColor="black"
        />
        <LabelText
          fontSize="small"
          isFixedWidth={true}
          isSpaceBetween={false}
          label="可提现时间："
          labelColType="medium"
          labelColor="grey"
          text="2019/08/13 16:38:43"
          textColor="black"
        />
      </div>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bottom": <div
          className="ljit-bank-card__bottom"
        >
          <Button
            color=""
            disabled={false}
            isFullWidth={true}
            onClick={[Function]}
            outline="solid"
          >
            解除绑定
          </Button>
        </div>,
        "className": "ljit-bank-card mock-class",
        "isShowingCloseButton": false,
        "left": <div
          className="ljit-bank-card__left"
        >
          <div
            className="ljit-bank-card__title"
          >
            mock-bankName
          </div>
          <LabelText
            fontSize="small"
            isFixedWidth={true}
            isSpaceBetween={false}
            label="持卡人："
            labelColType="medium"
            labelColor="grey"
            text="mock-payer"
            textColor="black"
          />
          <LabelText
            fontSize="small"
            isFixedWidth={true}
            isSpaceBetween={false}
            label="尾号："
            labelColType="medium"
            labelColor="grey"
            text="mock-number"
            textColor="black"
          />
        </div>,
        "onClose": [Function],
        "right": <div
          className="ljit-bank-card__right"
        >
          <LabelText
            fontSize="small"
            isFixedWidth={true}
            isSpaceBetween={false}
            label="绑定时间"
            labelColType="medium"
            labelColor="grey"
            text="2019/08/13 10:38:43"
            textColor="black"
          />
          <LabelText
            fontSize="small"
            isFixedWidth={true}
            isSpaceBetween={false}
            label="可提现时间："
            labelColType="medium"
            labelColor="grey"
            text="2019/08/13 16:38:43"
            textColor="black"
          />
        </div>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
